content/docs/guides/adopting/from_terraform.md

```diff
@@ -144,18 +148,92 @@ for i in range(2):
 {{% choosable language go %}}
 
 ```go
-// Terraform RemoteStateReference is not yet supported in Go.
-//
-// See https://github.com/pulumi/pulumi-terraform/issues/518.
+package main
+
+import (
+    "os"
+
+    "github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+    "github.com/pulumi/pulumi-terraform/sdk/v2/go/state"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+        cwd, err := os.Getwd()
+        if err != nil {
+            return err
+        }
+
+        state, err := state.NewRemoteStateReference(ctx, "localstate", &state.LocalStateArgs{
+            Path: pulumi.String(filepath.Join(cwd, "terraform.tfstate")),
+        })
+        if err != nil {
+            return err
+        }
+
+        publicSubnetIds := stateRef.Outputs.ApplyT(func(args interface{}) ([]string, error) {
+            ids := args.(map[string]interface{})["public_subnet_ids"].([]interface{})
+            subnetIds := make([]string, len(ids))
+            for i, v := range ids {
+                subnetIds[i] = v.(string)
+            }
+            return subnetIds, nil
+        }).(pulumi.StringArrayOutput)
+        for x := 0; x < 2; x++ {
+            ii := x
+            _, err := ec2.NewInstance(ctx, fmt.Sprintf("instance-%d", ii), &ec2.InstanceArgs{
+                Ami:          pulumi.String("ami-7172b611"),
+                InstanceType: pulumi.String("t2.medium"),
+                SubnetId:     publicSubnetIds.Index(pulumi.Int(ii)),
+            })
+            if err != nil {
+                return err
+            }
+        }
+
+        return nil
+	})
+}
 ```
 
 {{% /choosable %}}
 {{% choosable language csharp %}}
 
 ```csharp
-// Terraform RemoteStateReference is not yet supported in .NET.
-//
-// See https://github.com/pulumi/pulumi-terraform/issues/516.
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Pulumi;
+using Pulumi.Aws.Ec2;
+using Pulumi.Terraform.State;
+
+class Program
+{
+    static Task<int> Main()
+    {
+        return Deployment.RunAsync(() => {
+
+            var remoteState = new RemoteStateReference("localstate", new LocalBackendRemoteStateReferenceArgs
+            {
+                Path = Path.GetFullPath("terraform.tfstate"),
+            });
+
+            var subnetIds = remoteState.GetOutput("public_subnet_ids").
+                Apply(ids => ((ImmutableArray<object>) ids).Cast<string>().ToImmutableArray());
+
+            for (int i = 0; i < 2; i++)
+            {
+                var server = new Instance($"instance-{i}", new InstanceArgs
+                {
+                    Ami = "ami-7172b611",
+                    InstanceType = "t2.micro",
+                    SubnetId = subnetIds.GetAt(i),
+                });
+            }
+        });
```

Comment:
I thought we now encouraged a style in C# that uses a class inheriting from `Stack` - like https://github.com/pulumi/templates/blob/master/azure-csharp/MyStack.cs#L5.

Reply:
updated